app/src/main/java/hu/vmiklos/plees_tracker/MainActivity.kt

```diff
@@ -313,7 +315,20 @@ class MainActivity : AppCompatActivity(), View.OnClickListener {
         val startStop = findViewById<FloatingActionButton>(R.id.start_stop)
         val startStopText = findViewById<TextView>(R.id.start_stop_text)
 
+        val dndManager = getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+        val preferences = PreferenceManager.getDefaultSharedPreferences(applicationContext)
+        val dndEnabled = preferences.getBoolean("enable_dnd", false)
+
         if (DataModel.start != null && DataModel.stop != null) {
+            // When user stops tracking sleep
+            if (dndManager.isNotificationPolicyAccessGranted and dndEnabled) {
+                // Restore Do Not Disturb status when user started tracking
```

Comment:
This is a bit misleading,  you restore it when you stop, you just restore it to the value that was set when you started, right?